### ISSUE TITLE
Escape with CTRL+[

### ISSUE THREAD

**Issue by gerardroche** [NeoVintageous/NeoVintageous#197]
<a href="https://github.com/fullofcaffeine"><img src="https://avatars1.githubusercontent.com/u/81248?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [fullofcaffeine](https://github.com/fullofcaffeine)**
_Monday Nov 23, 2015 at 05:53 GMT_
_Originally opened as https://github.com/guillermooo/Vintageous/issues/1043_

----

I don't use ESC to go to the normal mode, but rather I've grown used to using `CTRL+[`, is there a way to set Vintageous to use it instead (or in addition to) ESC?

**Comment by gerardroche** [comment]
<a href="https://github.com/gerardroche"><img src="https://avatars0.githubusercontent.com/u/44148?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [gerardroche](https://github.com/gerardroche)**
_Monday Nov 23, 2015 at 20:16 GMT_

----

I think adding these keymaps will do it for you:

```
{"keys": ["ctrl+["], "command": "press_key",            "args": {"key": "<esc>"},           "context": [{"key": "vi_command_mode_aware"}]},
{"keys": ["ctrl+["], "command": "_enter_normal_mode",   "args": {"mode": "mode_insert"},    "context": [{"key": "vi_insert_mode_aware"}]},
```

**Comment by gerardroche** [comment]
<a href="https://github.com/ratiotile"><img src="https://avatars2.githubusercontent.com/u/7064626?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ratiotile](https://github.com/ratiotile)**
_Sunday Jan 24, 2016 at 22:06 GMT_

----

I use a different binding. What is the difference between `_enter_normal_mode` and `exit_insert_mode`?

```
{ "keys": ["ctrl+["], "command": "exit_insert_mode",
    "context":
    [
        { "key": "setting.command_mode", "operand": false },
        { "key": "setting.is_widget", "operand": false }
    ]
},
```

**Comment by gerardroche** [comment]
<a href="https://github.com/BjornMelgaard"><img src="https://avatars1.githubusercontent.com/u/7573215?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [BjornMelgaard](https://github.com/BjornMelgaard)**
_Thursday Dec 15, 2016 at 13:39 GMT_

----

Like @gerardroche, but with additions

```
  { "keys": ["ctrl+["], "command": "press_key",            "args": {"key": "<esc>"},        "context": [{"key": "vi_command_mode_aware"}]},
  { "keys": ["ctrl+["], "command": "_enter_normal_mode",   "args": {"mode": "mode_insert"}, "context": [{"key": "vi_insert_mode_aware"}]},
  { "keys": ["ctrl+["], "command": "clear_fields", "context":[  { "key": "has_next_field", "operator": "equal", "operand": true }] },
  { "keys": ["ctrl+["], "command": "clear_fields", "context":[  { "key": "has_prev_field", "operator": "equal", "operand": true }]},
  { "keys": ["ctrl+["], "command": "hide_panel", "args": {"cancel": true}, "context": [   { "key": "panel_visible", "operator": "equal", "operand": true } ]},
  { "keys": ["ctrl+["], "command": "hide_overlay", "context":[  { "key": "overlay_visible", "operator": "equal", "operand": true }]},
  { "keys": ["ctrl+["], "command": "hide_popup", "context":[  { "key": "popup_visible", "operator": "equal", "operand": true }]},
  { "keys": ["ctrl+["], "command": "hide_auto_complete", "context":[  { "key": "auto_complete_visible", "operator": "equal", "operand": true }]},
 
```

**Comment by gerardroche** [comment]
Duplciate of https://github.com/NeoVintageous/NeoVintageous/issues/249

**Comment by gerardroche** [comment]
@fullofcaffeine @ratiotile @BjornMelgaard This was fixed in NeoVintageous 1.1.0. get the latest version 1.4.4 with more bug fixes and features. NeoVintageous is a fork of the discontinued Vintageous plugin.
